src/drivers/tty/serial/ttys_processing.c

```diff
@@ -41,23 +41,32 @@ static inline struct uart *tty2uart(struct tty *tty) {
 	return tu->uart;
 }
 
-static int uart_rx_buff_put(struct uart *dev, int c) {
+static int uart_rx_buff_put(struct uart *dev) {
 	struct uart_rx *rx;
 
-	irq_lock();
-	{
-		rx = pool_alloc(&uart_rx_buff);
-		if (!rx) {
+	if (dev->tty) {
+		while (uart_hasrx(dev)) {
+			irq_lock();
+			{
+				rx = pool_alloc(&uart_rx_buff);
+				if (!rx) {
+					if (dev->uart_ops->uart_irq_dis) {
+						dev->uart_ops->uart_irq_dis(dev, &dev->params);
+						lthread_launch(&uart_rx_irq_handler);
```

Comment:
` lthread_launch(&uart_rx_irq_handler); ` have to called regardless of whether there is uart_irq_dis handler or not

Reply:
done